File: tests/e2e/types/requests/project-sprint/CreateProjectSprintRequest.ts
export default interface CreateProjectSprintRequest {
  projectId: string,
  name: string,
  startDate?: string | null,
  endDate?: string | null,
  keepPreviousStages?: boolean | undefined,
  projectStatus?: {
    title: string,
    status: string,
    description: string
  }
}

File: src/ConsiliumTempus.Api/Contracts/ProjectSprint/Create/CreateProjectSprintRequest.cs
using ConsiliumTempus.Domain.Project.Enums;

namespace ConsiliumTempus.Api.Contracts.ProjectSprint.Create;

public sealed record CreateProjectSprintRequest(
    Guid ProjectId,
    string Name,
    DateOnly? StartDate,
    DateOnly? EndDate,
    bool KeepPreviousStages,
    CreateProjectSprintRequest.CreateProjectStatus? ProjectStatus)
{
    public sealed record CreateProjectStatus(
        string Title,
        ProjectStatusType Status,
        string Description);
}